Gemfile

```
@@ -7,7 +7,7 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 gem 'rails', '~> 5.2.1'
 gem 'pg'
 # Use Puma as the app server
-gem 'puma', '~> 3.12'
+gem 'puma', '~> 4'
```

I'm pretty sure you want this:

```suggestion
gem 'puma', '~> 4.0'
```

---

[They're equivalent](https://guides.rubygems.org/patterns/#pessimistic_version_constraint):
  > It will only use the latest version from the 2.x series – so 2.3.0 – and not 3.0.0. 
unless I'm missing something else

---

Ah, you're right - sorry :) I just tested it with `gem install puma -v '~> 3'`, and it installed 3.12.4